llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -7834,6 +7843,56 @@ void SIInstrInfo::splitScalar64BitBFE(SIInstrWorklist &Worklist,
   addUsersToMoveToVALUWorklist(ResultReg, MRI, Worklist);
 }
 
+void SIInstrInfo::splitScalar64BitCountOp(SIInstrWorklist &Worklist,
```

Comment:
Can you add a comment showing what code this generates? I don't understand what the final min3 with 64 is for.

If the input it all zeros then S_FLBIT_I32_B64/S_FF1_I32_B64 would return -1, so this expansion should do the same.

Also are there any tests for this code? If not could you add some? See move-to-valu-* for examples.